src/main/java/com/example/Restaurant_Management_System_REST_API/DTO/CustomerDTOs/CustomerDTORequest.java

```diff
@@ -0,0 +1,42 @@
+package com.example.Restaurant_Management_System_REST_API.DTO.CustomerDTOs;
+
+import com.example.Restaurant_Management_System_REST_API.model.ContactDetails;
+import com.example.Restaurant_Management_System_REST_API.model.entity.Authority;
+import com.example.Restaurant_Management_System_REST_API.model.entity.Reservation;
+import jakarta.validation.constraints.Pattern;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor
+public class CustomerDTORequest {
+
+    private Long id;
+    private LocalDateTime creationTime;
+    private Reservation reservation;
+    private ContactDetails contactDetails;
+
+    //Below are fields for security issues
+    @Pattern(regexp = "^(?=.*[0-9])(?=.*[a-z])(?=.*[A-Z])(?=.*[@#$%^&+=])(?=\\S+$).{5,}$", message =
+            "Password must have at least 5 characters, one digit, one small letter, one big letter, and a special character")
+    private String password;
+    //            (?=.*[0-9]) ensures there is at least one digit.
+//            (?=.*[a-z]) ensures there is at least one lowercase letter.
+//            (?=.*[A-Z]) ensures there is at least one uppercase letter.
+//            (?=.*[@#$%^&+=]) ensures there is at least one special character.
+//            (?=\\S+$) ensures there are no spaces.
+//.           {5,} ensures there are at least 5 characters.
+    private Boolean accountNonExpired;//to avoid hardcoding I established these fields
+    private Boolean accountNonLocked;
+    private Boolean credentialsNonExpired;
+    private Boolean enabled;
+    private String emailAddress;
+    private Set<Authority> authorities;
```

Comment:
I'd sort those fields more accurately to their usage. By that I mean to place `private String emailAddress;` and `private Set<Authority> authorities;` above all the Boolean fields that you have commented as hardcoded. This will make the code more readable. It'd be very sexy to place distinct those hardcoded fields from all the others by making some comments in the code. This seriously improves code readability.

Reply:
allright! I have just moved String emailAddress and Set<Authority> authorities to the above but about the second part I must say that I have put those fields to avoid hardcoding, so they are not harcoded fields. 
They would be but in Customer class I set they return real values:

![image](https://github.com/BartoliniAndBorderCollies/Restaurant_Management_System_REST_API/assets/126821059/d388c79f-ac82-47a4-853f-539b97112aba)

Reply:
I didn't mean to move anything to other classes or I did not said nothing about changing the hardcoded part. I meant to rearrange lines of code (the fields part) in the class where I put the comment.

Reply:
I believe we didn't understood each other then :)